internal/provider/data_source_packer_image_iteration_test.go

```diff
@@ -0,0 +1,186 @@
+		case int(codes.AlreadyExists), http.StatusConflict:
+			// all good here !
+			return
+		}
+	}
+
+	t.Errorf("unexpected CreateIteration error, expected nil or 409. Got %v", err)
+}
+
+func getIterationIDFromFingerPrint(t *testing.T, bucketSlug, fingerprint string) string {
+	t.Helper()
+
+	client := testAccProvider.Meta().(*clients.Client)
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		OrganizationID: client.Config.OrganizationID,
+		ProjectID:      client.Config.ProjectID,
+	}
+
+	getItParams := packer_service.NewGetIterationParams()
+	getItParams.LocationOrganizationID = loc.OrganizationID
+	getItParams.LocationProjectID = loc.ProjectID
+	getItParams.BucketSlug = bucketSlug
+	getItParams.Fingerprint = &fingerprint
+
+	ok, err := client.Packer.GetIteration(getItParams, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return ok.Payload.Iteration.ID
+}
+
+func upsertBuild(t *testing.T, bucketSlug, fingerprint, iterationID string) {
+	client := testAccProvider.Meta().(*clients.Client)
+
+	createBuildParams := packer_service.NewCreateBuildParams()
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		OrganizationID: client.Config.OrganizationID,
+		ProjectID:      client.Config.ProjectID,
+	}
+	createBuildParams.LocationOrganizationID = loc.OrganizationID
+	createBuildParams.LocationProjectID = loc.ProjectID
+	createBuildParams.BucketSlug = bucketSlug
+	createBuildParams.BuildIterationID = iterationID
+
+	createBuildParams.Body = &models.HashicorpCloudPackerCreateBuildRequest{
+		Fingerprint: fingerprint,
+		BucketSlug:  bucketSlug,
+		Location:    loc,
+	}
+	createBuildParams.Body.Build = &models.HashicorpCloudPackerBuild{
+		PackerRunUUID: uuid.New().String(),
+		CloudProvider: "aws",
+		ComponentType: "amazon-ebs.example",
+		IterationID:   iterationID,
+		Status:        models.HashicorpCloudPackerBuildStatusDONE,
+		Images: []*models.HashicorpCloudPackerImage{
+			{
+				ImageID: "ami-42",
+			},
+			{
+				ImageID: "ami-43",
+			},
+		},
+	}
+	_, err := client.Packer.CreateBuild(createBuildParams, nil)
+	if err == nil {
+		return
+	}
+	if err, ok := err.(*packer_service.CreateBuildDefault); ok {
+		switch err.Code() {
+		case int(codes.AlreadyExists), http.StatusConflict:
+			// all good here !
+			return
+		}
+	}
+
+	t.Errorf("unexpected CreateBuild error, expected nil or 409. Got %v", err)
+}
+
+func TestAcc_dataSourcePacker(t *testing.T) {
+	resourceName := "data.hcp_packer_image_iteration.alpine"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t, false) },
+		ProviderFactories: providerFactories,
+
+		Steps: []resource.TestStep{
+			// testing that getting the production channel of the alpine image
+			// works.
+
+			{
+				PreConfig: func() {
+					bucket := "alpine"
+					fingerprint := "42"
+					upsertBucket(t, bucket)
+					upsertIteration(t, bucket, fingerprint)
+					itID := getIterationIDFromFingerPrint(t, bucket, fingerprint)
+					upsertBuild(t, bucket, fingerprint, itID)
```

Comment:
I've been getting an error when running this test:
```
{"error":"could not load the channel. something in the request is invalid. ","code":3,"message":"could not load the channel. something in the request is invalid. ","details":[]}
2021/08/05 12:05:49 [WARN] Got error running Terraform: exit status 1
```

I believe it's because the channel hasn't been upserted here.

Reply:
@azr this issue has been address in #180 with the addition of the createChannel and updateChannel calls.